src/ClpIrV1Decoder.cpp

```diff
@@ -0,0 +1,208 @@
+        {
+            auto const &log_event{m_log_events[log_event_idx]};
+            std::string message;
+
+            {
+                auto constant_handler = [&](std::string const &value, size_t begin_pos, size_t length)
+                {
+                    message.append(value, begin_pos, length);
+                };
+
+                auto encoded_int_handler = [&](clp::ir::four_byte_encoded_variable_t value)
+                { message.append(clp::ffi::decode_integer_var(value)); };
+
+                auto encoded_float_handler = [&](clp::ir::four_byte_encoded_variable_t encoded_float)
+                {
+                    message.append(clp::ffi::decode_float_var(encoded_float));
+                };
+
+                auto dict_var_handler = [&](std::string const &dict_var)
+                { message.append(dict_var); };
+
+                try
+                {
+                    clp::ffi::ir_stream::generic_decode_message<true>(
+                        log_event.get_logtype(),
+                        log_event.get_encoded_vars(),
+                        log_event.get_dict_vars(),
+                        constant_handler,
+                        encoded_int_handler,
+                        encoded_float_handler,
+                        dict_var_handler);
+                }
+                catch (clp::ffi::ir_stream::DecodingException const &e)
+                {
+                    std::cerr << "Failed to decode. Error Code: " << static_cast<uint32_t>(clp::ffi::ir_stream::IRErrorCode_Decode_Error)
+                              << std::endl;
+
+                    break;
+                }
+            }
+
+            LOG_VERBOSITY verbosity_level = LOG_VERBOSITY::NONE;
+            for (size_t verbosity_idx = 1; verbosity_idx < sizeof(VERBOSITIES) / sizeof(VERBOSITIES[0]); verbosity_idx++)
+            {
+                if (log_event.get_logtype().substr(1).starts_with(VERBOSITIES[verbosity_idx]))
+                {
+                    verbosity_level = static_cast<LOG_VERBOSITY>(verbosity_idx);
+                    break;
+                }
+            }
+
+            m_ts_pattern.insert_formatted_timestamp(
+                log_event.get_timestamp() + m_timezone_offset_ms,
+                message);
+
+            insert_row(results.as_handle(), message.c_str(), log_event.get_timestamp(),
+                       static_cast<int>(verbosity_level));
+        }
+
+        return true;
+    }
+
+private:
+    bool decodePreamble()
+    {
+        bool is_four_bytes_encoding{true};
+
+        if (auto const err{
+                clp::ffi::ir_stream::get_encoding_type(m_zstd_decompressor, is_four_bytes_encoding)};
+            clp::ffi::ir_stream::IRErrorCode::IRErrorCode_Success != err)
+        {
+            std::cout << "Failed to decode encoding type." << std::endl;
+            return false;
+        }
+        if (!is_four_bytes_encoding)
```

Comment:
```suggestion
        if (false == is_four_bytes_encoding)
```
We should apply yoda condition in C++ for two reasons:
1. Avoid implicty value assignment. A rvalue on the left hand side with `=` can't be compiled
2. Easier to read compared to `!condition`